app/main.py

```diff
@@ -1,34 +1,148 @@
+        self.save_field()
+        self._validate_field()
+
+    def save_field(self):
+        """This method assign a ship to every cell(e.g. deck)"""
+
+        for ship in self.ships:
+            ship_cell = Ship(ship[0], ship[1])
+            if ship[0][0] == ship[1][0]:
+                length = ship[1][1] - ship[0][1]
+                for idx in range(length + 1):
+                    self.field[(ship[0][0], ship[0][1] + idx)] \
+                        = ship_cell
+            elif ship[0][1] == ship[1][1]:
+                length = ship[1][0] - ship[0][0]
+                for idx in range(length + 1):
+                    self.field[(ship[0][0] + idx, ship[1][1])] \
+                        = ship_cell
 
     def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+        if location in self.field.keys():
+            self.field[location].fire(location[0], location[1])
+            if self.field[location].is_drowned:
+                return "Sunk!"
+            return "Hit!"
+        else:
+            return "Miss!"
+
+    def ships_on_field(self):
+        """This method create a matrix 10x10 aka field with ships
+        that corresponds to specific symbols. ~ - empty cell, □ - deck alive,
+        * - damaged ship, x - killed ship"""
+        new_field = [["~" for i in range(10)] for i in range(10)]
+        for deck in self.field.keys():
+            new_field[deck[0]][deck[1]] = u"\u25A1"
+            if not self.field[deck].get_deck(deck[0], deck[1]).is_alive:
+                new_field[deck[0]][deck[1]] = "*"
+                if self.field[deck].is_drowned:
+                    new_field[deck[0]][deck[1]] = "x"
+        return new_field
+
+    def print_field(self):
+        new_field = self.ships_on_field()
+        for ele in new_field:
+            print('\t'.join(ele))
+
+    @staticmethod
+    def distance(ship):
+        return math.sqrt((ship[1][0] - ship[0][0])**2
+                         + (ship[1][1] - ship[0][1])**2)
+
+    def _validate_field(self):
+        input_ships = sorted([self.distance(ship)
+                              + 1 for ship in self.ships], reverse=True)
+
+        if len(input_ships) != 10:
+            raise Exception("The total number of the ships should be 10!")
+        elif 4.0 not in input_ships:
+            raise Exception("There should be 1 four-deck ship!")
+        elif [3.0, 3.0] != input_ships[1:3]:
+            raise Exception("There should be 2 three-deck ship!")
+        elif [2.0, 2.0, 2.0] != input_ships[3:6]:
+            raise Exception("There should be 3 double-deck ship!")
+        elif [1.0, 1.0, 1.0, 1.0] != input_ships[6:]:
+            raise Exception("There should be 4 single-deck ship!")
+
+        """For each ship we book cells around and raise exception
```

Comment:
docstrings are not used inside the block if the code

Use regular comments for this purpose